modules/misc/test/tests/kernels/thermo_diffusion/tests

```diff
@@ -1,7 +1,31 @@
 [Tests]
-  [./test]
+  [./thermo_diffusion]
     type = 'Exodiff'
     input = 'thermo_diffusion.i'
     exodiff = 'thermo_diffusion_out.e'
+    requirement = 'Thermo diffusion shall reproduce an analytical solution'
+    design = "thermodiffusion.md"
+    issues = "#3392"
+  [../]
+  [./ad_thermo_diffusion]
+    type = 'Exodiff'
+    input = 'ad_thermo_diffusion.i'
+    exodiff = 'ad_thermo_diffusion_out.e'
+    requirement = 'AD thermo diffusion shall reproduce an analytical solution'
+    design = ""
+    issues = "#12633"
+    allow_test_objects = true
+  [../]
+  [./ad_thermo_diffusion_jacobian]
+    type = 'PetscJacobianTester'
+    input = 'ad_thermo_diffusion.i'
+    ratio_tol = 1e-7
+    difference_tol = 1e-3
```

Comment:
@lindsayad I'm not sure why this had to be loosened so much, it seems like the values are already low enough to signify a good jacobian?

Reply:
`ratio_tol` is a more important indicator of the Jacobian accuracy. If you have "larger" Jacobian values (e.g. a large Young's modulus in solid mechanics simulations without any manual scaling), then this `difference_tol` in general will need to be larger

Reply:
Okay, so if this is all okay, then this pr should be good to go for review